packages/flutter_tools/lib/src/fuchsia/fuchsia_dev_finder.dart

```diff
@@ -19,24 +22,42 @@ import 'fuchsia_sdk.dart';
 
 /// A simple wrapper for the Fuchsia SDK's 'device-finder' tool.
 class FuchsiaDevFinder {
+  FuchsiaDevFinder({
+    @required FuchsiaArtifacts fuchsiaArtifacts,
+    @required Logger logger,
+    @required ProcessManager processManager,
+  })
+    : _fuchsiaArtifacts = fuchsiaArtifacts,
+      _logger = logger,
+      _processUtils = ProcessUtils(logger: logger, processManager: processManager);
+
+
+  final FuchsiaArtifacts _fuchsiaArtifacts;
+  final Logger _logger;
+  final ProcessUtils _processUtils;
+
   /// Returns a list of attached devices as a list of strings with entries
   /// formatted as follows:
   /// 192.168.42.172 scare-cable-skip-joy
   Future<List<String>> list({ Duration timeout }) async {
-    if (fuchsiaArtifacts.devFinder == null ||
-        !fuchsiaArtifacts.devFinder.existsSync()) {
+    if (_fuchsiaArtifacts.devFinder == null ||
+        !_fuchsiaArtifacts.devFinder.existsSync()) {
       throwToolExit('Fuchsia device-finder tool not found.');
     }
     final List<String> command = <String>[
-      fuchsiaArtifacts.devFinder.path,
+      _fuchsiaArtifacts.devFinder.path,
       'list',
       '-full',
       if (timeout != null)
         ...<String>['-timeout', '${timeout.inMilliseconds}ms']
     ];
-    final RunResult result = await processUtils.run(command);
+    final RunResult result = await _processUtils.run(command);
     if (result.exitCode != 0) {
-      globals.printError('device-finder failed: ${result.stderr}');
+      // No devices returns error code 1.
+      // https://bugs.fuchsia.dev/p/fuchsia/issues/detail?id=48563
+      if (!result.stderr.contains('no devices found')) {
```

Comment:
This is the only behavior change, the rest of this diff is making `FuchsiaDevFinder` more testable.